src/compiler/checker.ts

```diff
@@ -12337,7 +12337,24 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function getNonMissingTypeOfSymbol(symbol: Symbol) {

```

Comment:
This is used by `isPropertySymbolTypeRelated`, the problem is that - at times - the instantiated mapped symbol is still deferred~ and is an indexed access type. When that happens, this function can't remove the missing type because it's not yet there.

To accommodate that, I clone the indexed access types resulting from that mapped symbol intantiation with a new `AccessFlags.NonMissing` so when the constraint of that indexed access gets finally requested the missing type can still be removed from it, like this code here intended to.